--- a/org-indent.el	2021-08-11 15:04:00.822293931 +0200
+++ b/org-indent.el	2021-08-31 17:02:08.912682469 +0200
@@ -126,6 +126,7 @@
 	(make-vector org-indent--deepest-level nil))
   (setq org-indent--text-line-prefixes
 	(make-vector org-indent--deepest-level nil))
+  (when (> org-indent-indentation-per-level 0)
   (dotimes (n org-indent--deepest-level)
     (let ((indentation (if (<= n 1) 0
 			 (* (1- org-indent-indentation-per-level)
@@ -150,7 +151,7 @@
 		(concat (make-string (+ n indentation) ?\s)
 			(and (> n 0)
 			     (char-to-string org-indent-boundary-char)))
-		nil 'face 'org-indent)))))
+		nil 'face 'org-indent))))))
 
 (defsubst org-indent-remove-properties (beg end)
   "Remove indentations between BEG and END."
